zebra-state/src/sled_state.rs

```diff
@@ -222,8 +323,13 @@ impl FinalizedState {
                     // for some reason type inference fails here
                     Ok::<_, sled::transaction::ConflictableTransactionError>(hash)
                 },
-            )
-            .map_err(Into::into)
+            );
+
+        if result.is_ok() {
+            self.stop_if_at_height_limit(StopCheckContext::OnCommit, height, hash);
```

Comment:
How do you feel about having `stop_if_at_height_limit` return a bool and flushing if it returns true, then we can remove the whole `StopCheckContext` enum.

Reply:
I think it's helpful to have the context in the log messages.

But I'm happy to inline the flush and exit into the callers, if you'd like.

Reply:
@yaahc I opened #1205 to follow up this refactor - let's discuss there?

I'd like to merge these tests soon, so we can make sure we don't break sync in other PRs.